MM-30335. Added DHCP scope migration docs.

rachelbritton opened the pull request:
Added a section about DHCP scope migration to the user manual.

rachelbritton committed:
MM-30335. Added DHCP scope migration docs.

Added a section about DHCP scope migration to the user manual.

rachelbritton committed:
MM-30335. Fixed typo.

rachelbritton committed:
MM-30335. Fixed typo.

rachelbritton committed:
MM-30807. Added info about scope lease time.

Added info about scope lease time setting conversion in migration from MS to ISC/Kea.

eythort-mm commented on guides/user-manual/dhcp_scopes.rst:
```diff
@@ -263,3 +263,115 @@ If a scope is no longer needed but you want to keep it for potential future use,
+
+Migrating Scopes
+----------------
+In the case that you need to decommission a server or there has been an outage, you can migrate a DHCP scope to a new server to keep your network operational.
```
Decomissioning or outage or not the only use cases - you may also just be migrating a scope to a server in a different location.

eythort-mm commented on guides/user-manual/dhcp_scopes.rst:
```diff
@@ -263,3 +263,115 @@ If a scope is no longer needed but you want to keep it for potential future use,
+In the case that you need to decommission a server or there has been an outage, you can migrate a DHCP scope to a new server to keep your network operational.
+
+.. note::
+  To migrate DHCP scopes, you must be a DHCP administrator with full read access to the source server(s) and scope(s), including reading options. You must also have permissions to create scopes on the destination server.
```
"DHCP administrator" indicates the user needs to have that built-in role in Micetro, but you only need the access rights referenced here.  Perhaps a "DHCP Editor" is softer ?

eythort-mm commented on guides/user-manual/dhcp_scopes.rst:
```diff
@@ -263,3 +263,115 @@ If a scope is no longer needed but you want to keep it for potential future use,
+.. note::
+  To migrate DHCP scopes, you must be a DHCP administrator with full read access to the source server(s) and scope(s), including reading options. You must also have permissions to create scopes on the destination server.
+
+Micetro not only allows you to migrate scopes between servers of the same type, but also between different server types, i.e., from a Microsoft DHCP server to an ISC server. Additionally, it's possible to migrate scopes from servers that are dead or unreachable. Refer to :ref:`migrate-unreachable-limitations` below.
```
Actually, we only support migration between specific server types for now.
- from MS DHCP to either ISC/MDDS or Kea
- from IOS to ISC/MDDS, Kea and MS

eythort-mm commented on guides/user-manual/dhcp_scopes.rst:
```diff
@@ -263,3 +263,115 @@ If a scope is no longer needed but you want to keep it for potential future use,
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+There are certain limitations to migrating scopes between servers of differing types:
+
+* DHCP policies are MS-specific and are not migrated, i.e., by converting to client classes on Kea servers.
```
perhaps leave out the i.e. ?

eythort-mm commented on guides/user-manual/dhcp_scopes.rst:
```diff
@@ -263,3 +263,115 @@ If a scope is no longer needed but you want to keep it for potential future use,
+* Reservations inside pools on MS servers are not migrated unless the setting to **Allow reservations inside pools on ISC DHCP servers** is enabled.
+* User class options are MS-specific and are not migrated.
+* DDNS settings are MS-specific and are not migrated.
+* When migrating scopes from an MS server to an ISC or Kea server, the MS DHCP option 51 (scope lease time) becomes the scope settings ``default-lease-time`` and ``max-lease-time``. 
```
option 51 is converted to default-lease-time/max-lease-time on ISC/MDDS and to valid-lifetime and max-lifetime on Kea

eythort-mm commented on guides/user-manual/dhcp_scopes.rst:
```diff
@@ -263,3 +263,115 @@ If a scope is no longer needed but you want to keep it for potential future use,
+
+* If the server is a MS DHCP server and a backup exists (the **Perform backup of MS and ISC DHCP servers** system setting is enabled), DHCP options are migrated from the backup for both scope and reservations.
+* DHCP policies and their options are not migrated.
+* Scope settings on ISC and Kea servers are not migrated.
```
probably scope configuration, like discussed privately?

eythort-mm commented on guides/user-manual/dhcp_scopes.rst:
```diff
@@ -263,3 +263,115 @@ If a scope is no longer needed but you want to keep it for potential future use,
+    .. image:: ../../images/migrate-dhcp-scope.png
+      :width: 80%
+
+  * **Source server**: Use the dropdown to select the server from which you want to migrate the scope.
```
It might be worth mentioning that when selecting multiple scopes, the dropdown is populated with the servers that are common amongst the scopes that are selected, i.e. only servers which all the selected scopes are on

eythort-mm commented on guides/user-manual/dhcp_scopes.rst:
```diff
@@ -263,3 +263,115 @@ If a scope is no longer needed but you want to keep it for potential future use,
+
+5. If there are no errors or warnings---or you want to proceed despite a warning---select :guilabel:`Migrate`.
+
+A progress bar is displayed in the dialog box, which shows you how much is left to migrate. When the progress bar reaches 100%, a ta-dah! sound indicates that the migration is complete.
```
the sound is not played :)
That was a joke by Örvar :)

rachelbritton replied:
I fully fell for it!

rachelbritton committed:
MM-30335. Revised text.

Revised text based on feedback and added info about which server types scopes can be migrated to/from other server types.
